components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -174,6 +200,86 @@ private boolean validateUserCredentials(OAuth2AccessTokenReqDTO tokenReq, Servic
+     * @param authenticated Boolean value which determines whether the user is authenticated or not.
+     * @param serviceProvider Service provider which contains the details of the application.
+     */
+    private void publishAuthenticationData(OAuth2AccessTokenReqDTO tokenReq, boolean authenticated,
+                                           ServiceProvider serviceProvider) {
+
+        AuthenticatedUser authenticatedUser = getAuthenticatedUser(tokenReq, serviceProvider);
+        AuthenticationContext authenticationContext = initializeAuthContext(authenticatedUser, serviceProvider);
+        AuthenticationDataPublisher authnDataPublisherProxy =
+                OAuth2ServiceComponentHolder.getAuthenticationDataPublisherProxy();
+        if (authnDataPublisherProxy != null && authnDataPublisherProxy.isEnabled(authenticationContext)) {
+            Map<String, Object> paramMap = new HashMap<>();
+            paramMap.put(FrameworkConstants.AnalyticsAttributes.USER, authenticatedUser);
+            Map<String, Object> unmodifiableParamMap = Collections.unmodifiableMap(paramMap);
+            if (authenticated) {
+                authnDataPublisherProxy
+                        .publishAuthenticationStepSuccess(null, authenticationContext, unmodifiableParamMap);
+                authnDataPublisherProxy.publishAuthenticationSuccess(null, authenticationContext, unmodifiableParamMap);
+            } else {
+                authnDataPublisherProxy.
+                        publishAuthenticationStepFailure(null, authenticationContext, unmodifiableParamMap);
+                authnDataPublisherProxy.publishAuthenticationFailure(null, authenticationContext,
+                        unmodifiableParamMap);
+            }
+        }
+    }
+
+    /**
+     * This method will create an AuthenticationContext object which needs to be passed to the publish methods.
+     *
+     * @param authenticatedUser User which tries to be authenticate.
+     * @param serviceProvider Service provider which contains the details of the application.
+     * @return An AuthenticationContest object with relevant details.
+     */
+    private AuthenticationContext initializeAuthContext(AuthenticatedUser authenticatedUser,
+                                                        ServiceProvider serviceProvider) {
+
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        String contextId = UUIDGenerator.generateUUID();
+        authenticationContext.setContextIdentifier(contextId);
+        authenticationContext.setTenantDomain(authenticatedUser.getTenantDomain());
+        authenticationContext.setRequestType(OAUTH2);
+        authenticationContext.setRememberMe(false);
+        authenticationContext.setForceAuthenticate(true);
+        authenticationContext.setPassiveAuthenticate(false);
+        authenticationContext.setProperty(IS_INITIAL_LOGIN, true);
+
+        // setting sequenceConfig with authenticatedUser, serviceProvider
```

Comment:
Start with a capital letter and end with a full stop for the comments. Do this in all applicable places.

Reply:
Modified